lib/theory/src/Rule.hs

```diff
@@ -46,6 +46,17 @@ equalOpenRuleUpToDiffAnnotation (OpenProtoRule ruE1 ruAC1) (OpenProtoRule ruE2 r
   equalRuleUpToDiffAnnotationSym ruE1 ruE2 && length ruAC1 == length ruAC2 &&
   all (uncurry equalRuleUpToDiffAnnotationSym) (zip ruAC1 ruAC2)
 
+detectReadOnly :: Rule i -> Rule i
```

Comment:
Please comment what this function does. (In particular as detect sounds passive, but it modifies the rule)

Reply:
I removed it as it was only for the automatic conversion from `F -> F` to `!F -> ..` which we decided to not use.